### DISCUSSION TITLE
state reloading problem / how to force the state of my HomePage to change

### DISCUSSION THREAD

**Discussion by geonidas6** [nylo-core/nylo#133] [Q&A]
I need your help please
What I want to do is simply change the color and opacity of a container when a condition is true.
the varible values change but not the view.
but I used reboot()

here's an extract of the code and a screenshot of the interface

` Row(
                                mainAxisAlignment:
                                MainAxisAlignment.spaceBetween,
                                children: [
                                  for (int index = 0;
                                  index < listOfGenders.length;
                                  index++)
                                    Material(
                                      elevation: 5,
                                      borderRadius:
                                      BorderRadius.circular(10),
                                      child: InkWell(
                                        onTap: () {
                                          selectGender(index);
                                          print(index);
                                          reboot();

                                        },
                                        child: Container(
                                          width: size.width * 0.275,
                                          height: size.width * 0.4,
                                          decoration: BoxDecoration(
                                            borderRadius:
                                            BorderRadius.circular(10),
                                            border: Border.all(
                                              width: 2,
                                              color:
                                              selectedGenderIndex ==
                                                  index
                                                  ? colorScheme
                                                  .primary
                                                  .withOpacity(
                                                  0.7)
                                                  : colorScheme
                                                  .tertiary,
                                            ),
                                            color: selectedGenderIndex ==
                                                index
                                                ? colorScheme.primary
                                                .withOpacity(0.8)
                                                : colorScheme.tertiary,
                                          ),
                                          child: Column(
                                            mainAxisAlignment:
                                            MainAxisAlignment.center,
                                            children: [
                                              Image.asset(
                                                listOfGenders[index]
                                                ["icon"]!,
                                                width: size.width * 0.2,
                                                height: size.width * 0.2,
                                              ).localAsset(),
                                              const Gap(5),
                                              Text(
                                                listOfGenders[index]
                                                ["name"]!,
                                                style: TextStyle(
                                                  color:
                                                  selectedGenderIndex ==
                                                      index
                                                      ? Colors.white
                                                      : Colors.white
                                                      .withOpacity(
                                                      0.5),
                                                  fontWeight:
                                                  FontWeight.bold,
                                                  fontSize: 16,
                                                ),
                                              ),
                                            ],
                                          ),
                                        ),
                                      ),
                                    ),
                                ])`
                                
                                
                                `  void selectGender(int index) {
    setState(() {

      selectedGenderIndex = index;
    });
  }`
                                
                                
                                
![Screenshot_20240428_195840](https://github.com/nylo-core/nylo/assets/38140825/3d75e92f-f33c-4a02-a23d-a60abd1205df)

**Comment by agordn52** [comment]
Hi @geonidas6,

No problem, I don't think you need to use `reboot` as this will call the `boot` method again.

Instead, just use `setState` and also for the Modal Bottom Sheet use `StatefulBuilder`, this will allow you to update the state of the Modal Bottom Sheet.

https://stackoverflow.com/questions/52414629/how-to-update-state-of-a-modalbottomsheet-in-flutter

Hope that helps.

**Reply by geonidas6** [reply]
Hello ,
Thank you so much👍 for your solution. I tested it and it worked.

I have the same problem on the login form.

although the value of the variable _obscureText changes but the password is not displayed it remains obscure.

can you help me?


`NyTextField(

              decoration: InputDecoration(

                border: OutlineInputBorder(

                  borderRadius: BorderRadius.all(Radius.circular(30.0)), 

                ),

                enabledBorder: OutlineInputBorder(

                  borderSide: BorderSide(color: BLACK21),

                  borderRadius: BorderRadius.all(Radius.circular(30.0)), 

                ),

                focusedBorder: OutlineInputBorder(

                  borderSide: BorderSide(color: BLACK21),

                  borderRadius: BorderRadius.all(Radius.circular(30.0)),

                ),
                labelText: Str.upper(trans("mot_de_passe")),

                labelStyle: TextStyle(color: _color2),

                 suffixIcon: IconButton(

                    icon: Icon(_iconVisible, color: Colors.grey[400], size: 20),

                    onPressed: () {

                      _toggleObscureText();

                    }),

              ),
              controller: textFieldPassword,

              labelText: Str.upper(trans("mot_de_passe")),

              obscureText: _obscureText,

              dummyData: "",

            ),`




`void _toggleObscureText() {
setState(() {

      _obscureText = !_obscureText;

      if (_obscureText == true) {

        _iconVisible = Icons.visibility_off;

      } else {

        _iconVisible = Icons.visibility;

      }

    });

  }`

**Reply by geonidas6** [reply]
Please sorry can you help me. or tell me what I'm doing wrong?

**Reply by agordn52** [reply]
Hi @geonidas6,

I'll take a look at my earliest convenience.

**Reply by geonidas6** [reply]
Okay, thank you in advance

**Reply by geonidas6** [reply]
There is another problem. I just noticed that when the validation function validate It's in the controller it doesn't work. And there is no error displayed. But when I put it directly in LoginPage it works.
